### PR TITLE
fix: Update copy when other team has guest links disabled SQSERVICES-1439

### PR DESCRIPTION
**Pull request by agisilaos**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/SQSERVICES-1439" title="SQSERVICES-1439" target="_blank">BugSQSERVICES-1439</a>  [iOS] Update copy when other team has guest links disabled and I look into guest link options
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

With this PR we update the copy for when other team has guest links disabled to match what web is doing on their end. We need to handle two different scenarios and present the correct copy. 

In the first case if the conversation is from the Self Team we're going to show: 

`Generating guest links is not allowed in your team.`

In the second case if the conversation is from the other team we're going to show:

`You can't generate a guest link in this conversation, as it has been created by someone from another team and this team is not allowed to use guest links`

Snapshot Tests have been updated to cover both of those cases.

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: Wire-iOS/Sources/UserInterface/Conversation Options/GuestLinkInfoCell.swift [hunk 1]
```diff
@@ -64,7 +64,6 @@ final class GuestLinkInfoCell: UITableViewCell, CellConfigurationConfigurable {
             label.topAnchor.constraint(equalTo: contentView.topAnchor),
             label.trailingAnchor.constraint(equalTo: contentView.trailingAnchor),
             label.bottomAnchor.constraint(equalTo: contentView.bottomAnchor),
-            label.heightAnchor.constraint(equalToConstant: 56)
```
Comment (agisilaos) [inline Wire-iOS/Sources/UserInterface/Conversation Options/GuestLinkInfoCell.swift:67]: Removed the height constraint so we can fit the new copy for the string without any issue.

**Review by KaterinaWire — APPROVED** [review]

File: Wire-iOS/Generated/Strings+Generated.swift [hunk 2]
```diff
@@ -2101,7 +2101,7 @@ internal enum L10n {
           internal static let title = L10n.tr("Localizable", "guest_room.link.header.title")
         }
         internal enum NotAllowed {
-          /// Generating guest links is not allowed in your team.
+          /// You can't generate a guest link in this conversation, as it has been created by someone from another team and this team is not allowed to use guest links.
```
Comment (johnxnguyen) [inline Wire-iOS/Generated/Strings+Generated.swift:2104]: Should this new copy be _in addittion_ to the existing copy? Guest links may by disabled for a conversation for two reasons:

1. It's disabled for your team
2. It's disabled for another team, to which the conversation belongs to.

Both of these cases warrant separate copy, so I think we need to keep the existing error messages and distinguish between the two when setting the error message.

**Comment by KaterinaWire** [comment]
@agisilaos Shouldn't we have snapshot tests for both cases?

File: Wire-iOS/Sources/UserInterface/Conversation Options/ConversationGuestOptionsViewModel.swift [hunk 3]
```diff
@@ -132,7 +132,7 @@ final class ConversationGuestOptionsViewModel {
             return rows
         case .disabled:
             rows.append(.linkHeader)
-            rows.append(.info)
+            rows.append(.info(isSelfTeam: true))
```
Comment (KaterinaWire) [inline Wire-iOS/Sources/UserInterface/Conversation Options/ConversationGuestOptionsViewModel.swift:135]: Shouldn't we somehow check that this is selfTeam?
I didn't find any other code where this value is different.

File: Wire-iOS/Sources/UserInterface/Conversation Options/GuestLinkInfoCell.swift [hunk 4]
```diff
@@ -63,24 +62,25 @@ final class GuestLinkInfoCell: UITableViewCell, CellConfigurationConfigurable {
...
 
     func configure(with configuration: CellConfiguration, variant: ColorSchemeVariant) {
-        guard case .info = configuration else { preconditionFailure() }
+        guard case let .info(isSelfTeam) = configuration else { preconditionFailure() }
         accessibilityIdentifier = "guest_links.not_allowed.cell"
         iconImageView.setIcon(.about, size: .tiny, color: UIColor.from(scheme: .textForeground, variant: variant))
         label.textColor = UIColor.from(scheme: .textForeground, variant: variant)
+
+        let infoTextForSelfTeam = L10n.Localizable.GuestRoom.Link.NotAllowed.ForSelfTeam.explaination
+        let infoTextForOtherTeam = L10n.Localizable.GuestRoom.Link.NotAllowed.ForOtherTeam.explaination
```
Comment (KaterinaWire) [inline Wire-iOS/Sources/UserInterface/Conversation Options/GuestLinkInfoCell.swift:78]: I think we can create typealias here:
typealias GuestRoomStrings = L10n.Localizable.GuestRoom.Link.NotAllowed

or maybe there is a better name

File: Wire-iOS/Generated/Strings+Generated.swift (hunk 2 above)
Comment (agisilaos) [reply Wire-iOS/Generated/Strings+Generated.swift]: @johnxnguyen Covered all the cases above. PR is ready for review again.

File: Wire-iOS/Sources/UserInterface/Conversation Options/ConversationGuestOptionsViewModel.swift (hunk 3 above)
Comment (agisilaos) [reply Wire-iOS/Sources/UserInterface/Conversation Options/ConversationGuestOptionsViewModel.swift]: @KaterinaWire This is outdated. Please check the latest changes.

**Comment by agisilaos** [comment]
> @agisilaos Shouldn't we have snapshot tests for both cases?

We do now. Please check the latest changes.

File: Wire-iOS/Sources/UserInterface/Conversation Options/GuestLinkInfoCell.swift (hunk 4 above)
Comment (agisilaos) [reply Wire-iOS/Sources/UserInterface/Conversation Options/GuestLinkInfoCell.swift:77]: Hmm, nice catch will think about it and come back with a better name.

File: Wire-iOS/Sources/UserInterface/Conversation Options/Conversation+OptionsConfiguration.swift [hunk 5]
```diff
@@ -55,6 +55,18 @@ extension ZMConversation {
...
+            let selfUser = ZMUser.selfUser(inUserSession: userSession)
+            let selfUserTeamId = selfUser.teamIdentifier
+
+            if conversation.teamRemoteIdentifier == selfUserTeamId {
+                return true
+            } else {
+                return false
+            }
+
+        }
```
Comment (KaterinaWire) [inline Wire-iOS/Sources/UserInterface/Conversation Options/Conversation+OptionsConfiguration.swift:68]:
```suggestion
            var isConversationFromSelfTeam: Bool {
             let selfUser = ZMUser.selfUser(inUserSession: userSession)
             
             return conversation.teamRemoteIdentifier == selfUser.teamIdentifier
```

Comment (KaterinaWire) [inline Wire-iOS/Sources/UserInterface/Conversation Options/Conversation+OptionsConfiguration.swift:68]:
```suggestion
        var isConversationFromSelfTeam: Bool {
            let selfUser = ZMUser.selfUser(inUserSession: userSession)
            let selfUserTeamId = selfUser.teamIdentifier

            if conversation.teamRemoteIdentifier == selfUserTeamId {
                return true
            } else {
                return false
            }

        }
```

Comment (agisilaos) [reply Wire-iOS/Sources/UserInterface/Conversation Options/Conversation+OptionsConfiguration.swift:68]: Fixed it with the latest commit Katerina!

File: Wire-iOS/Sources/UserInterface/Conversation Options/GuestLinkInfoCell.swift (hunk 4 above)
Comment (agisilaos) [reply Wire-iOS/Sources/UserInterface/Conversation Options/GuestLinkInfoCell.swift]: @KaterinaWire This is fixed with my latest commit. Let me know what you think.

File: Wire-iOS/Sources/UserInterface/Conversation Options/Conversation+OptionsConfiguration.swift (hunk 5 above)
Comment (KaterinaWire) [reply Wire-iOS/Sources/UserInterface/Conversation Options/Conversation+OptionsConfiguration.swift:68]: What do you mean? There's previous code.

**Review by KaterinaWire — APPROVED** [review]

File: Wire-iOS/Sources/UserInterface/Conversation Options/Conversation+OptionsConfiguration.swift (hunk 5 above)
Comment (KaterinaWire) [reply Wire-iOS/Sources/UserInterface/Conversation Options/Conversation+OptionsConfiguration.swift]: Yes

File: Wire-iOS/Resources/Base.lproj/Localizable.strings [hunk 6]
```diff
@@ -1374,7 +1374,8 @@
 
 "guest_room.link.header.title" = "Guest Links";
 "guest_room.link.header.subtitle" = "Invite others with a link to this conversation. Anyone with the link can join the conversation, even if they don’t have Wire.";
-"guest_room.link.not_allowed.explaination" = "Generating guest links is not allowed in your team.";
+"guest_room.link.not_allowed.for_self_team.explaination" = "Generating guest links is not allowed in your team.";
+"guest_room.link.not_allowed.for_other_team.explaination" = "You can't generate a guest link in this conversation, as it has been created by someone from another team and this team is not allowed to use guest links.";
```
Comment (johnxnguyen) [inline Wire-iOS/Resources/Base.lproj/Localizable.strings:1378]:
```suggestion
"guest_room.link.not_allowed.for_self_team.explanation" = "Generating guest links is not allowed in your team.";
"guest_room.link.not_allowed.for_other_team.explanation" = "You can't generate a guest link in this conversation, as it has been created by someone from another team and this team is not allowed to use guest links.";
```

**Review by johnxnguyen — APPROVED** [review]

File: Wire-iOS/Resources/Base.lproj/Localizable.strings (hunk 6 above)
Comment (agisilaos) [reply Wire-iOS/Resources/Base.lproj/Localizable.strings]: Fixed it.

**Review by Matausi29 — COMMENTED** [review]
[](https://github.com/desktop/desktop/projects/8#card-20660292)

**Pull request merged**